lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -673,9 +681,12 @@ class QcFlagService {
 
             if (bad) {
                 runSummary[QC_SUMMARY_PROPERTIES.badEffectiveRunCoverage] = parseFloat(effectiveRunCoverage);
-                runSummary[QC_SUMMARY_PROPERTIES.mcReproducible] = mcReproducible;
+                runSummary[QC_SUMMARY_PROPERTIES.mcReproducible] =
+                    mcReproducible || runSummary[QC_SUMMARY_PROPERTIES.mcReproducible];
```

Comment:
These looks very similar to the way you compute coverage per detector, do you think you can factorize? It starts to have quite some business logic and it looks like you always have to change both at the same time...

Reply:
factorized